spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/tracing/otlp/OtlpAutoConfiguration.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2012-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.actuate.autoconfigure.tracing.otlp;
+
+import java.util.Map.Entry;
+
+import io.micrometer.tracing.otel.bridge.OtelTracer;
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.exporter.otlp.http.trace.OtlpHttpSpanExporter;
+import io.opentelemetry.exporter.otlp.http.trace.OtlpHttpSpanExporterBuilder;
+import io.opentelemetry.sdk.trace.SdkTracerProvider;
+
+import org.springframework.boot.actuate.autoconfigure.tracing.ConditionalOnEnabledTracing;
+import org.springframework.boot.autoconfigure.AutoConfiguration;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+
+/**
+ * {@link EnableAutoConfiguration Auto-configuration} for OTLP. Brave does not support
+ * OTLP, so we only configure it for OpenTelemetry. OTLP defines three transports that are
+ * supported: gRPC (/protobuf), HTTP/protobuf, HTTP/JSON. From these transports HTTP/JSON
+ * is not supported by the OTel Java SDK, and it seems there are no plans supporting it in
+ * the future, see: <a href=
+ * "https://github.com/open-telemetry/opentelemetry-java/issues/3651">opentelemetry-java#3651</a>.
+ * Because this class configures components from the OTel SDK, it can't support HTTP/JSON.
+ * To keep things simple, we only auto-configure HTTP/protobuf. If you want to use gRPC,
+ * please disable this auto-configuration and create a bean.
+ *
+ * @author Jonatan Ivanov
+ * @since 3.1.0
+ */
+@AutoConfiguration
+@ConditionalOnEnabledTracing
+@ConditionalOnClass({ OtelTracer.class, SdkTracerProvider.class, OpenTelemetry.class, OtlpHttpSpanExporter.class })
+@ConditionalOnProperty(prefix = "management.otlp.tracing", name = "enabled", matchIfMissing = true)
```

Comment:
We try to avoid `….enabled` properties if we can. Do we really need once here? FWIW, we don't have one for Zipkin.

Reply:
I would still very much prefer the arrangement you @wilkinsona suggested in https://github.com/spring-projects/spring-boot/pull/34508#discussion_r1129226532.

It feels like single exporter use case is the dominant one, and it's not like one exporter option is much more popular than the other one - AFAIK gRPC is still the default with OTel agent. So while I agree that HTTP exporter is the better default going forward, IMO auto-configuration should also provide value for those that prefer (or have to use) gRPC.

Reply:
@wilkinsona Thanks, fixed.

@vpavic gRPC might be the default for the OTel agent but I think there are a few things here we need to consider (apart the fact that the agent is not used here):
- As far as I remember for the non-agent use-case HTTP/protobuf is the default: https://github.com/open-telemetry/opentelemetry-java/pull/3786.
- The OTel Java SDK used to use gRPC by default but there were lots of issues: users running into dependency conflicts and they needed to bring and configure their own gRPC lib eventually. See https://github.com/spring-projects-experimental/spring-cloud-sleuth-otel/issues/39 and https://github.com/open-telemetry/opentelemetry-java/issues/3123#issuecomment-814577392.
- Since OTel SDKs do not need to support all three transports, the OTel Collectors must (to support all SDKs) which means no-one is forced (by the collectors) to use a certain transport. In fact, the Java SDK does not support HTTP/JSON and it is not planned that it will be ever supported (See [CNCF Slack](https://cloud-native.slack.com/archives/C014L2KCTE3/p1677780415787389)).

Because of these I'm tending towards not adding support for gRPC and let the users configure it if they want to. We can always revisit this if there will be user need for gRPC (FWIW, we did not see this ask with Micrometer's OTLP registry).

Reply:
I missed the fact that `io.opentelemetry/opentelemetry-exporter-otlp` doesn't pull in gRPC dependencies (other dependency in my current project was pulling them in so just I wrongly assumed) and that of course changes things a bit and goes more in favor of providing auto-configuration for only HTTP exporter.